AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java

```diff
@@ -1969,8 +1969,8 @@ private void restoreCollectionPreferences() {
             // Dynamic don't have review options; attempt to get deck-specific auto-advance options
             // but be prepared to go with all default if it's a dynamic deck
             JSONObject revOptions = new JSONObject();
-            if (!getCol().getDecks().isDyn(getCol().getDecks().current().getLong("id"))) {
-                revOptions = getCol().getDecks().confForDid(getCol().getDecks().current().getLong("id")).getJSONObject("rev");
+            if (!getCol().getDecks().isDyn(getCol().getDecks().selected())) {
+                grevOptions = getCol().getDecks().confForDid(getCol().getDecks().selected()).getJSONObject("rev");
```

Comment:
Seems fine, I just think 'grevOptions' was a typo and you meant the following?

```suggestion
                revOptions = getCol().getDecks().confForDid(getCol().getDecks().selected()).getJSONObject("rev");
```